src/v/cloud_storage_clients/client_pool.cc

```diff
@@ -58,7 +68,27 @@ client_pool::acquire(ss::abort_source& as) {
             if (_policy == client_pool_overdraft_policy::wait_if_empty) {
                 co_await _cvar.wait();
             } else {
-                _pool.emplace_back(make_client());
+                // Ask other shards for allowance
```

Comment:
this else branch is taken for the `create_new_if_empty` policy, which we do not use right now anywhere. Are we going to switch to this policy in a future PR?

Reply:
I renamed the policy into `borrow_if_empty` and enabled in in `application.cc`